lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -85,82 +91,369 @@ describe('NxTable', function() {
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveAccessibleName('asdf');
+    });
+
+    it('sets the isFilterHeader flag into the RowContext, normalized to a boolean', function() {
+      function ContextDependentChild() {
+        const { isFilterHeader } = useContext(RowContext);
+        return <span data-testid="context-dependent">{isFilterHeader.toString()}</span>;
+      }
+
+      const { rerender } = render(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow isFilterHeader={true}>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('true');
+
+      rerender(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow isFilterHeader={false}>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+
+      rerender(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow isFilterHeader={null}>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+
+      rerender(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+    });
+  });
+
+  describe('NxTableCell', function() {
+    const renderHeaderComponent = (extraProps?: NxTableCellProps) => render(
+      <NxTable>
+        <NxTableHead>
+          <NxTableRow>
+            <HeaderContext.Provider value={true}>
```

Comment:
The `HeaderContext` is also an implementation detail, and definitely should never be used in this way - the provider is set up in `NxTable.Head`

Reply:
> The `HeaderContext` is also an implementation detail, and definitely should never be used in this way - the provider is set up in `NxTable.Head`

Thanks! I adjusted it here: 09f61036e296efa83f8cce271d5ba1ed82bc5463